helpers/paragraph.js

```diff
@@ -181,4 +181,33 @@ export function findEndOfActivePartOfNote(note: TNote): number {
         : lineCount)
   // console.log(`  dHL = ${doneHeaderLine}, cHL = ${cancelledHeaderLine} endOfActive = ${endOfActive}`)
   return endOfActive
-}
\ No newline at end of file
+}
+
+/**
+ * @description Function to map a TParagraph to a SimpleParagraph, to only have content and type
+ * @example note.paragraphs.map(paragraphToSimpleParagraphMapper(())
+ * @author @m1well
+ *
+ * @returns {function(TParagraph): {type: ParagraphType, content: string}|null}
+ */
+export const paragraphToSimpleParagraphMapper = (): Function => {
```

Comment:
This is interesting. I'm curious why this function returns a function. Why did this need to be a composition?

Reply:
because then you can use it directliy in the `map` part:
```
yesterdayNote.paragraphs
          .filter(para => para.type === 'open')
          .map(paragraphToSimpleParagraphMapper()) <----- here
          .map(simpleParagraphScheduledFromYesterday()) <----- this is the same principle
```

also the same principle like my sorting I suggested also in the plugin-dev channel:
```
export const sortByPrio = (): Function => {
  return (second: TParagraph, first: TParagraph) => {
    if (haveSameTypes(first, second)) {
      if (isFlaggedThrice(first) || isFlaggedThrice(second)) {
        return isFlaggedThrice(first) ? 1 : -1
      }
      if (isFlaggedTwice(first)) {
        return isFlaggedThrice(second) ? -1 : 1
      }
      if (isFlaggedTwice(second)) {
        return isFlaggedThrice(first) ? 1 : -1
      }
      if (isFlaggedOnce(first)) {
        return (isFlaggedTwice(second) || isFlaggedThrice(second)) ? -1 : 1
      }
      if (isFlaggedOnce(second)) {
        return (isFlaggedTwice(second) || isFlaggedThrice(first)) ? 1 : -1
      }
    }
    return 0
  }
}
```
call:
```
doneTasks
            .sort(sortByPrio()) <---- here
            .map(cleanupParagraphs())
```

so this can avoid to always write functions which has an array as input parameter, do something, and then return the changed array ;)